Lib/compileall.py

```diff
@@ -70,13 +69,21 @@ def compile_dir(dir, maxlevels=10, ddir=None, force=False, rx=None,
     workers:   maximum number of parallel workers
     invalidation_mode: how the up-to-dateness of the pyc will be checked
     """
-    if workers is not None and workers < 0:
-        raise ValueError('workers must be greater or equal to 0')
-
+    global ProcessPoolExecutor
+    if workers is not None:
+        if workers < 0:
+            raise ValueError('workers must be greater or equal to 0')
+        elif workers > 1 and ProcessPoolExecutor is None:
+            try:
+                from concurrent.futures import ProcessPoolExecutor as PPE
+            except ImportError:
```

Comment:
I don't think tri-state global variable is needed.  It can be more simple:

```py
if workers > 1:
    try:
        from concurrent.futures import ProcessPoolExecutor
    except ImportError:
        workers = 1
```

Reply:
The problem with doing this is that if the import fails, since this path occurs per-file, then you'll be doing a failed import over and over again. That seems like a Bad Thing.